### PR TITLE
Update win_domain_group_membership.ps1

### PR DESCRIPTION
**Pull request by AravindBalajiS**
Bug fixes while Adding and Deleting a group from another group

##### SUMMARY

I was trying to add DEF group to ABC group after executing the ansible script I checked the members that are present in ABC groups, the DEF group was not added to it. As DEF group was not present in ABC group after executing the ansible script then I tried to add DEF group to ABC group using the ps1 script, after execution of the ps1 script I found that DEF is a universal group and ABC is a global group, as a global group cannot have a universal group as a member. So the ps1 script failed while executing. But while executing the ansible script it got passed, it didn't show any error while execting it.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

win_domain_group_membership
##### ADDITIONAL INFORMATION

**STEPS TO REPRODUCE**
This can be reproduced by running the following playbook

```
---
- hosts: nodes
  vars:
    domain_user_name: '************************'                                                           # Service Account Username
    domain_user_password: '********************'                                                           # Service Account Password                                                                         
    domain_name: '**********************'                                                                      # Zone of the DNS

  tasks:
  - name: Add a domain group to a domain group
    win_domain_group_membership:
      name: "ABC"
      members:
        - "DEF"
      domain_username: "{{ domain_user_name }}"
      domain_password: "{{ domain_user_password }}"
      domain_server: "{{ domain_name }}"
      state: present
```
**EXPECTED RESULTS**

```
TASK [Add a domain group to a domain group] *************************************************************************************************************************************************************************************************
task path: /home/test.yml:12
Using module file /usr/lib/python2.7/site-packages/ansible/modules/windows/win_domain_group_membership.ps1
<xx.xx.xx.xx> ESTABLISH WINRM CONNECTION FOR USER: test on PORT 5986 TO xx.xx.xx.xx
checking if winrm_host xx.xx.xx.xx is an IPv6 address
EXEC (via pipeline wrapper)
fatal: [xx.xx.xx.xx]: FAILED! => {
    "added": [],
    "changed": false,
    "msg": "Failed to add a group CN=DEF: A global group cannot have a universal group as a member",
    "removed": []
}
```
**ACTUAL RESULTS**

```
TASK [Add a domain group to a domain group] *************************************************************************************************************************************************************************************************
task path: /home/test.yml:12
Using module file /usr/lib/python2.7/site-packages/ansible/modules/windows/win_domain_group_membership.ps1
<xx.xx.xx.xx> ESTABLISH WINRM CONNECTION FOR USER: test on PORT 5986 TO xx.xx.xx.xx
checking if winrm_host xx.xx.xx.xx is an IPv6 address
EXEC (via pipeline wrapper)
changed: [xx.xx.xx.xx] => {
    "added": [
        "DEF"
    ],
    "changed": true,
    "members": [
        "XYZ"
    ],
    "removed": []
}
```

```paste below

```

### CODE REVIEW

**Comment by ShachafGoldstein** [comment]
Please add a changelog fragment to document the bug fix.

**Comment by AravindBalajiS** [comment]
I have added the changelog fragment [57104](https://github.com/ansible/ansible/pull/57104)

**Comment by AravindBalajiS** [comment]
Hi @ShachafGoldstein 
Any update?

File: lib/ansible/modules/windows/win_domain_group_membership.ps1 [hunk 1]
```diff
@@ -70,11 +70,19 @@ foreach ($member in $members) {
     }
 
     if ($state -in @("present", "pure") -and !$user_in_group) {
-        Add-ADGroupMember -Identity $name -Members $group_member -WhatIf:$check_mode @extra_args
+        try {
+            Add-ADGroupMember -Identity $name -Members $group_member -WhatIf:$check_mode @extra_args
+        } catch {
+            Fail-Json $result "Failed to add a group $($group_member): $($_.Exception.Message)"
```
Comment (ShachafGoldstein) [inline lib/ansible/modules/windows/win_domain_group_membership.ps1:76]: A silly comment but probably should be `Failed to add group $($group_member): $($_.Exception.Message)` or `Failed to add a group: $($group_member) - $($_.Exception.Message)`

File: lib/ansible/modules/windows/win_domain_group_membership.ps1 [hunk 2]
```diff
@@ -70,11 +70,19 @@ foreach ($member in $members) {
...
+        }
         $result.added.Add($group_member.SamAccountName)
         $result.changed = $true
     } elseif ($state -eq "absent" -and $user_in_group) {
-        Remove-ADGroupMember -Identity $name -Members $group_member -WhatIf:$check_mode @extra_args -Confirm:$False
+        try {
+            Remove-ADGroupMember -Identity $name -Members $group_member -WhatIf:$check_mode @extra_args -Confirm:$False
+        } catch {
+            Fail-Json $result "Failed to remove a group $($group_member): $($_.Exception.Message)"
```
Comment (ShachafGoldstein) [inline lib/ansible/modules/windows/win_domain_group_membership.ps1:84]: Same as the add comment

File: lib/ansible/modules/windows/win_domain_group_membership.ps1 [hunk 3]
```diff
@@ -94,7 +102,11 @@ if ($state -eq "pure") {
         }
 
         if ($user_to_remove) {
-            Remove-ADGroupMember -Identity $name -Members $current_member -WhatIf:$check_mode @extra_args -Confirm:$False
+            try {
+                Remove-ADGroupMember -Identity $name -Members $current_member -WhatIf:$check_mode @extra_args -Confirm:$False
+            } catch {
+                Fail-Json $result "Failed to remove a group $($group_member): $($_.Exception.Message)"
```
Comment (ShachafGoldstein) [inline lib/ansible/modules/windows/win_domain_group_membership.ps1:108]: Here as well, regarding the grammar/text

**Comment by ShachafGoldstein** [comment]
> Hi @ShachafGoldstein
> Any update?

I can't really merge anything but if it is urgent for you, you can add it to the weekly meeting agenda and join the IRC to move it faster.

**Comment by jborean93** [comment]
Personally I don’t see the point to this PR. A try/catch is only fired in the event of a terminating error which means the script would have stopped. Since 2.8 we should be giving better error message that will result in a lot better error output which removes the necessity of the Fail-Json for individual commands. We should only be using Fail-Json if we are manually throwing a failure or the error message doesn’t make sense in the context where it happens.

**Comment by AravindBalajiS** [comment]
@ShachafGoldstein @jborean93 
Thanks for comments

**Comment by gundalow** [comment]
Given https://github.com/ansible/ansible/pull/56953#issuecomment-504699181 I think this can be closed.

**Pull request closed without merging**